peregrine/analysis/run_sim.py

```diff
@@ -0,0 +1,266 @@
+#!/usr/bin/env python
+
+# Copyright (C) 2016 Swift Navigation Inc.
+# Contact: Perttu Salmela <psalmela@exafore.com>
+#
+# This source is subject to the license found in the file 'LICENSE' which must
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+
+import os
+import sys
+import argparse
+import subprocess
+import json
+import datetime
+import numpy
+import peregrine.iqgen.bits.signals as signals
+import numpy as np
+
+IQ_DATA = "iqdata.bin"
+TRACK_DATA = "track_res"
+TRACK_RES_DATA = "track_res.json"
+
+
+def runCmd(cmd):
+  print cmd
+  # Do not use shell=True for untrusted input!
+  out = subprocess.check_output(cmd, shell=True)
+  print out
+  return out
+
+
+def runIqGen(lens, snr, dop, acc):
+  cmd = "python " + peregrinePath() + "/peregrine/iqgen/iqgen_main.py"
+  cmd = cmd + " --gps-sv 1 --encoder 1bit --bands l1ca+l2c --message-type crc --profile low_rate"
+  if float(acc) != 0.0:
+    cmd = cmd + " --doppler-type linear --doppler-speed " + acc + " "
+  elif float(dop) != 0.0:
+    cmd = cmd + " --doppler-type const " + dop + " "
+  else:
+    cmd = cmd + " --doppler-type zero "
+  cmd = cmd + " --snr " + snr + " --generate " + lens + " "
+  cmd = cmd + " --output " + IQ_DATA + " "
+  out = runCmd(cmd)
+  lines = out.split('\n')
+  cn0 = None
+  l2dop = None
+  l2chip = None
+  for ln in lines:
+    words = ln.split()
+    if len(words) == 0:
+      continue
+    if words[0] == ".L2":
+      cn0 = words[2]
+    if words[0] == ".l2_doppler:":
+      l2dop = words[1]
+    if words[0] == ".l2_chip:":
+      l2chip = words[1]
+    if words[0] == ".SNR":
+      if float(words[2]) != float(snr):
+        print "snr unexpected"
+        sys.exit(1)
+  if cn0 == None or l2dop == None or l2chip == None:
```

Comment:
Should be
condition **is** None

http://stackoverflow.com/questions/3257919/is-none-vs-none